### ISSUE TITLE
Upgrade to cup 0.11b

### ISSUE THREAD

**Issue by regisd** [jflex-de/jflex#175]
CUP has released version 0.11b
- CUP version 0.11b 20140808 is a feature/bugfix release.
  - added expected symbols to the report_error(...) method
  - added an expected_token_ids() to inform about valid continuations
  - added parameters and code to enable automatic creation of XML parse trees
  - fixed a bug in the lookahead computation
- CUP version 0.11b beta 20140226 is a feature release.
  - Command line flag -locations now triggers the generation of the identifiers
  - xleft and xright in the actions section instead of left and right.
  - Both are of type ComplexSymbolFactory.Location instead of plain ints.
- CUP version 0.11b beta 20140220 is a feature release.
  - Main switch statement now splits the action code in bundles of 300 productions
    on several different method bodies. Should fix the 65k size boundary problem

**Comment by lsf37** [comment]
Should we use it for the 1.6.1 release or leave the update for after that?

**Comment by tbl0605** [comment]
For me, latest dev version of JFlex 1.6.1 seems rock solid and corrects a long standing bug (#135) blocking us migrating from version 1.4.3, and should be released asap ;) So why introduce now potentially new problems by changing the cup version ? Of course, it's only my opinion! :)

Btw, there's even a newer cup version (0.11b 20150226) available on http://www2.cs.tum.edu/projects/cup/, but without any changelog.

**Comment by lsf37** [comment]
I'm tending to leave it for the next release as well. Any objections?

**Comment by regisd** [comment]
I agree.

**Comment by tbl0605** [comment]
Hello,
I recently discovered that version 11b-20150226 had a nasty bug (introduced somewhere after v10k) that was corrected in latest 11b-20150326 version. In the changelog of cup it's described as :
"Fixed an issue with empty productions having non-empty location information".
It's about wrong position calculation of empty productions.

I was the original reporter of this bug to the maintainer of CUP.
I don't know if it impacts JFlex, but nevertheless, I still strongly advice you to make an additional update (to 11b-20150326), only a few files changed ;)
